### PR TITLE
GRP-1091 - Change the display properties of groups from within loader…

### PR DESCRIPTION
**Pull request by sachdeva-vivek**
… for addIncludeExclude and requireInGroups. Also change the display properties for systemOfRecord groups.

### CODE REVIEW

File: grouper/src/grouper/edu/internet2/middleware/grouper/Group.java [hunk 1]
```diff
@@ -31,6 +31,20 @@
 */
 
 package edu.internet2.middleware.grouper;
+import static edu.internet2.middleware.grouper.hooks.examples.GroupTypeTupleIncludeExcludeHook.excludeDescription;
```
Comment (mmoayyed) [inline grouper/src/grouper/edu/internet2/middleware/grouper/Group.java]: Wouldnt you just want to do:

import static edu.internet2.middleware.grouper.hooks.examples.GroupTypeTupleIncludeExcludeHook.* ?

**Comment by sachdeva-vivek** [comment]
Made the change. Actually, import list was collapsed in Eclipse so did not notice it while writing code. Thanks.

File: grouper/src/grouper/edu/internet2/middleware/grouper/Group.java [hunk 2]
```diff
@@ -7627,5 +7633,234 @@ public boolean canHavePrivilege(Subject subject, String privilegeOrListName, boo
...
+    if (!alreadyChanged && this.getName().endsWith(systemOfRecordExtensionSuffix()) && groupExistsAsCompositeInOverallGroup()) {
+        DisplayProperties displayProperties = new DisplayProperties();
+        changeDisplayPropertiesForIncludeExcludeType(displayProperties);
+        alreadyChanged = true;
+    }
+    
+    // requireInGroups type from UI - systemOfRecord group is not of type requireInGroups
+    if (!alreadyChanged && this.getName().endsWith(systemOfRecordExtensionSuffix()) && groupExistsAsMemberInOverallGroup()) {
+      DisplayProperties displayProperties = new DisplayProperties();
+      changeDisplayPropertiesForRequireInGroupsType(displayProperties);
```
Comment (mmoayyed) [inline grouper/src/grouper/edu/internet2/middleware/grouper/Group.java]: Is this block missing ` alreadyChanged = true; `?

**Comment by sachdeva-vivek** [comment]
Not really, it is the last if block and it is outside of loop.

**Comment by mchyzer** [comment]
Vivek, do you think there could be a problem is the user renaming a group doesnt have ADMIN on all the other groups?  What will happen, an error and tx rolled back?  Wondering if this should run as grouper system so we dont have that problem (if its not already)

**Comment by sachdeva-vivek** [comment]
Chris, yes I think there will be a problem when the display name is changed from UI. I will test it and let you know what happens. I will change it to run as grouper session if I find any issues. Thanks for the good catch.

**Pull request merged**